RAT/inputs.py

```diff
@@ -0,0 +1,296 @@
+                           for param in getattr(project, class_list) if not param.fit]
+    problem.otherLimits = [[param.min, param.max] for class_list in RAT.project.parameter_class_lists
+                           for param in getattr(project, class_list) if not param.fit]
+
+    return problem
+
+
+def make_cells(project: RAT.Project) -> Cells:
+    """Constructs the cells input required for the compiled RAT code.
+
+    Note that the order of the inputs (i.e, f1 to f20) has been hard--coded into the compiled RAT code.
+
+    Parameters
+    ----------
+    project : RAT.Project
+        The project model, which defines the physical system under study.
+
+    Returns
+    -------
+    cells : RAT.rat_core.Cells
+        The set of inputs that are defined in MATLAB as cell arrays.
+    """
+
+    hydrate_id = {'bulk in': 1, 'bulk out': 2}
+
+    # Set contrast parameters according to model type
+    if project.model == Models.StandardLayers:
+        if project.calculation == Calculations.Domains:
+            contrast_models = [[project.domain_contrasts.index(domain_contrast, 1) for domain_contrast in contrast.model]
+                               for contrast in project.contrasts]
+        else:
+            contrast_models = [[project.layers.index(layer, 1) for layer in contrast.model]
+                               for contrast in project.contrasts]
+    else:
+        contrast_models = [[]] * len(project.contrasts)
+
+    # Get details of defined layers
+    layer_details = []
+    for layer in project.layers:
+
+        layer_params = []#[project.parameters.index(getattr(layer, attribute), 1) for attribute in list(layer.model_fields.keys())[1:-2]]
+        layer_params.append(project.parameters.index(layer.hydration, 1) if layer.hydration else float('NaN'))
+        layer_params.append(hydrate_id[layer.hydrate_with])
+
+        layer_details.append(layer_params)
+
+    # Find contrast data in project.data classlist
+    all_data = []
+    data_limits = []
+    simulation_limits = []
+
+    for contrast in project.contrasts:
+
+        data_index = project.data.index(contrast.data)
+
+        if 'data' in project.data[data_index].model_fields_set:
+            all_data.append(project.data[data_index].data)
+            data_limits.append(project.data[data_index].data_range)
+            simulation_limits.append(project.data[data_index].simulation_range)
+        else:
+            all_data.append([0.0, 0.0, 0.0])
+            data_limits.append([0.0, 0.0])
+            simulation_limits.append([0.0, 0.0])
+
+    # Populate the set of cells
+    cells = Cells()
+    cells.f1 = [[0, 1]] * len(project.contrasts)  # This is marked as "to do" in RAT
+    cells.f2 = all_data
+    cells.f3 = data_limits
+    cells.f4 = simulation_limits
+    cells.f5 = [contrast_model if contrast_model else 0 for contrast_model in contrast_models]
+    #cells.f6 = layer_details if project.model == Models.StandardLayers else [0]
```

Comment:
why the comment?

Reply:
This code, and the line noted above, are commented out because they cause the code to crash, as noted here: https://github.com/RascalSoftware/RAT/issues/221 I need a response to this so I can act on it appropriately. It was always my intention to have this sorted before merging this PR. As far as I can see, what is required here is to ensure that layers can only be specified with a full set of parameters, meaning that we remove some code from the matlab and make the fields of the layer model required arguments here.